### PR TITLE
Added smarter medic logic, pumped up and resurrection policy, helper functions

### PR DESCRIPTION
**Pull request by ghost**
Added the rest of the abilities to ITEMS dictionary, and sorted it.
Created helper functions to get the percentage of your total team in a lane, and to get the number of an ability active in a lane. Helpful in deciding if an ability would have the greatest impact in a lane.

Created revive ability policy, reviving if the lane has > 20% of total players and > 33% of the lane players are dead. Some sites incorrectly say revive affects all lanes, but thorough testing says otherwise.
Modified medic policy to also look at the team's overall health (just looking at the 10 buckets of the histogram) and react even if the player's health is above 50%. Medic policy now looks at pumped up ability first, as it is basically a superior medic.

Things that could be tweaked/changed:
- Include reflect shield in medic policy. Not sure if it's a better god mode because their 'reflect' seems like you still take damage
- Resurrect policy might be extended to attempt respawn, since resurrect will revive the player as well.
- 20% of total players, 33% of the lane players, 40% of health bins used in medic policy, and bins 1-6 in medic policy are all arbitrary and should be discussed
- Better distribution of the aforementioned bins (i.e. bin1 \* 0.9 + bin2 \* 0.8 etc.) when deciding to use on team
- Not sure if visibility is checked when triggering items and abilities?

### CODE REVIEW

**Comment by mouseas** [comment]
I need 2 people to independently test this for errors and functionality.

**Comment by Page-** [comment]
```
Uncaught SyntaxError: Unexpected token =
    at Object.InjectedScript._evaluateOn (<anonymous>:895:140)
    at Object.InjectedScript._evaluateAndWrap (<anonymous>:828:34)
    at Object.InjectedScript.evaluate (<anonymous>:694:21)
```

**Comment by ghost** [comment]
@Page- Do you know what was being run at the time?

**Comment by Page-** [comment]
I used https://github.com/JCKE/steamSummerMinigame/blob/f500cbd3c558257c9f91678f9fecf3d2fe1ed265/autoPlay.js directly in the chrome console - sadly it mangles the lines so I couldn't comment directly on the offending line (not sure which one it is)

**Comment by ghost** [comment]
@Page- In the console, beside the error in the top-right corner, it should have some sort of token error, like VM12345:6 or something. If you don't see it, you may have to click the arrow to expand the error. If you click on the tokens, one of them should take you to the spot in the code that caused the error.

**Comment by Page-** [comment]
Duh, yeah, been a while since I've debugged console run code:

```
Uncaught SyntaxError: Unexpected token =
    at Object.InjectedScript._evaluateOn (<anonymous>:895:140)
    at Object.InjectedScript._evaluateAndWrap (<anonymous>:828:34)
    at Object.InjectedScript.evaluate (<anonymous>:694:21)
InjectedScript._evaluateOn @ VM276:895
InjectedScript._evaluateAndWrap @ VM276:828
InjectedScript.evaluate @ VM276:694
```

**Comment by Page-** [comment]
None of them go to anything useful though :S

**Comment by ghost** [comment]
@Page- Aha, the problem seems to be with this function:

``` javascript
function numAbilityInLane(abilityId, lane=g_Minigame.CurrentScene().m_nExpectedLane) {
    // Checks if an ability is in use in a given lane and returns the number
    var numActive = g_Minigame.CurrentScene().m_rgLaneData[lane].abilities[abilityId];
    if (numActive === undefined)
        return 0;
    return numActive;
}
```

Testing in Firefox, it was fine defining an optionable parameters like this, but apparently it's the wrong way. This has basically been a crash course in JavaScript for me, so I'm not really surprised to find this haha

**Comment by mouseas** [comment]
Aaaand today's game is done. Looks like testing will have to halt for an hour or so.

**Comment by ghost** [comment]
Darn! Well, I tested it a bit before the shutdown, so I'll commit that. In other news, my badge auto purchase system is almost ready!

**Comment by win0na** [comment]
You may want to add the LICENSE back.

File: LICENSE [hunk 1]
```diff
@@ -1,22 +0,0 @@
-The MIT License (MIT)
```
Comment (mouseas) [inline LICENSE:1]: Need to add the license file back in.

**Comment by win0na** [comment]
Is this ready for merge? If so, I can fix the conflicts easily.

**Comment by ghost** [comment]
I hope so; after testing it for awhile it has yet to break on me.

**Pull request closed without merging**